src/curves/bip32Ed25519.ts

```diff
@@ -0,0 +1,94 @@
+import { pbkdf2 } from '@noble/hashes/pbkdf2';
+import { sha512 } from '@noble/hashes/sha512';
+import libsodium from 'libsodium-wrappers-sumo';
+
+export const curve = ed25519.CURVE;
+
+/**
+ * Named after whitepaper: BIP32-Ed25519 Hierarchical Deterministic Keys over a Non-linear Keyspace
+ * https://input-output-hk.github.io/adrestia/static/Ed25519_BIP.pdf
+ */
+export const name = 'bip32Ed25519';
+
+// Secret is empty string if not provided by user
+export const secret = stringToBytes('');
+
+/**
+ * Always returns true.
+ * For root node derivation, https://github.com/cardano-foundation/CIPs/blob/09d7d8ee1bd64f7e6b20b5a6cae088039dce00cb/CIP-0003/Icarus.md does not mention any cases when the private key is invalid.
+ * For child node derivation, https://input-output-hk.github.io/adrestia/static/Ed25519_BIP.pdf does not mention any cases when the private key is invalid.
+ *
+ * @param _privateKey - The private key to check.
+ * @returns True.
+ */
+export const isValidPrivateKey = (_privateKey: Uint8Array | string | bigint) =>
+  true;
+
+export const deriveUnhardenedKeys = true;
+
+export const publicKeyLength = 32;
+
+export const getPublicKey = async (
+  privateKey: Uint8Array,
+  _compressed?: boolean,
+): Promise<Uint8Array> => {
+  await libsodium.ready;
+  return libsodium.crypto_scalarmult_ed25519_base_noclamp(
+    // TODO: imo we should be able to replace this with noble
+    privateKey.slice(0, 32),
+  );
+};
+
+export const publicAdd = (
+  _publicKey: Uint8Array,
+  _tweak: Uint8Array,
+): Uint8Array => {
+  throw new Error('Not supported.');
+};
+
+export const compressPublicKey = (publicKey: Uint8Array): Uint8Array => {
+  // Ed25519 public keys don't have a compressed form.
+  return publicKey;
+};
+
+export const decompressPublicKey = (publicKey: Uint8Array): Uint8Array => {
+  // Ed25519 public keys don't have a compressed form.
+  return publicKey;
+};
+
+// Consistent with the Icarus derivation scheme.
+// https://github.com/cardano-foundation/CIPs/blob/09d7d8ee1bd64f7e6b20b5a6cae088039dce00cb/CIP-0003/Icarus.md
+const clampScalar = (scalar: Uint8Array): Uint8Array => {
+  /* eslint-disable no-bitwise */
+  scalar[0] &= 0b1111_1000;
+  scalar[31] &= 0b0001_1111;
+  scalar[31] |= 0b0100_0000;
+  return scalar;
+};
+
+export const privateKeyLength = 64;
+
+export const rootKeyPairIsCreatedFrom = 'entropy';
+
+/**
+ * Converts entropy to a root key pair.
+ * This function is consistent with the Icarus derivation scheme.
+ * Icarus root key derivation scheme: https://github.com/cardano-foundation/CIPs/blob/09d7d8ee1bd64f7e6b20b5a6cae088039dce00cb/CIP-0003/Icarus.md.
+ * CIP#: https://github.com/cardano-foundation/CIPs/blob/09d7d8ee1bd64f7e6b20b5a6cae088039dce00cb/CIP-0003/CIP-0003.md#master-key-generation.
+ *
+ * @param entropy - The entropy value.
+ * @returns The root key pair consisting of 64-byte private key and 32-byte chain code.
+ */
+export const entropyToRootKeyPair = async (
```

Comment:
I'd consider adding explicitly "Icarus" to the name of the function, otherwise it can become quite misleading because one may legitimately thing the function does the same as seedFromRootKeyPair, just transforms entropy to seed on top which would be completely wrong and could leat to fatal consequences

```suggestion
export const entropyToIcarusRootKeyPair = async (
```